File: hw/tipantiza/u3/hw21FarmSystemPOO/FarmSystem_py/controller/i_meat_animal.py
from abc import ABC, abstractmethod
from typing import List
from model.cut import Cut
from model.slaughter_house import SlaughterHouse

class IMeatAnimal(ABC):
    @abstractmethod
    def cut(self) -> List[Cut]:
        pass
    
    @abstractmethod
    def send_to_slaughter_house(self, slaughter_house: SlaughterHouse):
        pass
    
    @abstractmethod
    def is_ready_for_slaughter(self) -> bool:
        pass

File: hw/tipantiza/u3/hw21FarmSystemPOO/FarmSystem_js/controller/i-meat-animal.js
export class IMeatAnimal {
    cut() {
        throw new Error('Method cut() must be implemented');
    }
    
    sendToSlaughterHouse(slaughterHouse) {
        throw new Error('Method sendToSlaughterHouse() must be implemented');
    }
    
    isReadyForSlaughter() {
        throw new Error('Method isReadyForSlaughter() must be implemented');
    }
}